components/brave_wallet_ui/components/extension/connect-with-site-panel/connect-with-site-panel.tsx

```diff
@@ -0,0 +1,275 @@
+  ScrollContainer,
+  BackgroundContainer,
+  SelectAddressContainer,
+  ButtonRow,
+  PermissionsWrapper,
+  PermissionsContainer,
+  SectionLabel,
+  SectionPoint,
+  BulletContainer,
+  BulletIcon,
+  AddAccountText,
+  AddAcountIcon,
+  IconCircle,
+  WhiteSpace
+} from './connect-with-site-panel.style'
+import {
+  ConnectPanelButton,
+  AccountNameText
+} from './select-account-item/select-account-item.style'
+import {
+  Row,
+  HorizontalSpace,
+  VerticalSpace,
+  VerticalDivider
+} from '../../shared/style'
+
+// Utils
+import { getLocale } from '../../../../common/locale'
+
+interface Props {
+  originInfo: SerializableOriginInfo
+  accountsToConnect: WalletAccountType[]
+}
+
+export const ConnectWithSite = (props: Props) => {
+  const { originInfo, accountsToConnect } = props
+
+  // Redux
+  const dispatch = useDispatch()
+
+  // State
+  const [addressToConnect, setAddressToConnect] = React.useState<string>()
+  const [selectedDuration, setSelectedDuration] =
+    React.useState<BraveWallet.PermissionLifetimeOption>(
+      BraveWallet.PermissionLifetimeOption.kPageClosed
+    )
+  const [isReadyToConnect, setIsReadyToConnect] = React.useState<boolean>(false)
+  const [isScrolled, setIsScrolled] = React.useState<boolean>(false)
+
+  // Refs
+  let scrollRef = React.useRef<HTMLDivElement | null>(null)
+
+  // Methods
+  const onNext = React.useCallback(() => {
+    if (!isReadyToConnect) {
+      setIsReadyToConnect(true)
+      return
+    }
+    if (addressToConnect) {
+      dispatch(
+        PanelActions.connectToSite({
+          addressToConnect: addressToConnect,
+          duration: selectedDuration
+        })
+      )
+    }
+  }, [isReadyToConnect, addressToConnect, selectedDuration])
+
+  const onCancel = React.useCallback(() => {
+    dispatch(PanelActions.cancelConnectToSite())
+  }, [])
+
+  const onSelectAccount = React.useCallback(
+    (account: WalletAccountType) => () => {
+      if (addressToConnect === account.address) {
+        setAddressToConnect(undefined)
+        return
+      }
+      setAddressToConnect(account.address)
+    },
+    [addressToConnect]
+  )
+
+  const onClickAddAccount = React.useCallback(() => {
+    chrome.tabs.create(
+      { url: `chrome://wallet${WalletRoutes.AddAccountModal}` },
+      () => {
+        if (chrome.runtime.lastError) {
+          console.error(
+            'tabs.create failed: ' + chrome.runtime.lastError.message
+          )
+        }
+      }
+    )
+  }, [])
+
+  const onScroll = () => {
+    const scrollPosition = scrollRef.current
+    if (scrollPosition !== null) {
+      const { scrollTop } = scrollPosition
+      if (scrollTop > 40) {
+        setIsScrolled(true)
+      } else {
+        setIsScrolled(false)
+      }
+    }
+  }
+
+  return (
+    <StyledWrapper>
+      <BackgroundContainer
+        backgroundImage={`chrome://favicon/size/64@1x/${originInfo.originSpec}`}
+      />
+
+      <ConnectWithSiteHeader
+        isScrolled={isScrolled}
```

Comment:
Follow up to polish the transition

Reply:
Ditto on issue created: https://github.com/brave/brave-browser/issues/29299